float64[] x_array
float64[] y_array
